x-pack/legacy/plugins/siem/public/containers/case/use_post_case.tsx

```diff
@@ -4,24 +4,25 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Dispatch, SetStateAction, useEffect, useReducer, useState } from 'react';
+import { useReducer, useCallback } from 'react';
+
+import { CaseRequest } from '../../../../../../plugins/case/common/api';
 import { useStateToaster } from '../../components/toasters';
 import { errorToToaster } from '../../components/ml/api/error_to_toaster';
-import * as i18n from './translations';
-import { FETCH_FAILURE, FETCH_INIT, FETCH_SUCCESS, POST_NEW_CASE } from './constants';
-import { Case, NewCase } from './types';
+
 import { createCase } from './api';
-import { getTypedPayload } from './utils';
+import { FETCH_FAILURE, FETCH_INIT, FETCH_SUCCESS } from './constants';
+import * as i18n from './translations';
+import { Case } from './types';
 
 interface NewCaseState {
-  data: NewCase;
-  newCase?: Case;
+  caseData: Case | null;
   isLoading: boolean;
   isError: boolean;
 }
 interface Action {
   type: string;
-  payload?: NewCase | Case;
+  payload?: Case;
```

Comment:
I want to change all these actions to be like I did in use_get_cases:

```
export type Action =
  | { type: 'FETCH_FAILURE' }
  | { type: 'FETCH_INIT' }
  | { type: 'FETCH_SUCCESS'; payload: Case };
```

Reply:
I can do in a follow up PR

Reply:
make sense to me